backend/libraries/gated_groups/src/lib.rs

```diff
@@ -136,6 +145,21 @@ fn check_verified_credential_gate(
     CheckIfPassesGateResult::Success
 }
 
+fn check_composite_gate_synchronously(gate: CompositeGate, args: CheckGateArgs) -> CheckIfPassesGateResult {
+    let count = gate.inner.len();
+    for (index, inner) in gate.inner.into_iter().enumerate() {
+        let result = check_if_passes_gate_synchronously(inner, args.clone());
```

Comment:
Given the types, this ends up being recursive (as a composite gate can contain composite gates) so we should probably have definite termination to that recursion (even if we don't anticipate the front end creating recursive gates) like a max depth?

Reply:
I've added validation to disallow any composite gates within a composite gate so it shouldn't ever happen but yeah I'll see if I can protect against that in a nice way here too.

Reply:
Updated. It'll be fairly easy to switch this to allow some nesting if we ever decide to do so.